[mio] error linking executable related to lib mio after updating my vcpkg

**Describe the bug**
I updated my vcpkg today to this commit: https://github.com/microsoft/vcpkg/commit/871e20cf67ebaa80c048b868be53c8e7375c79f3
The problem started here: d235330027cacdfffc25cc0001eeecb0694e84ed

And I started to have a linking problem related to library "mio".
https://github.com/mandreyel/mio

There was no relevant change in the Canary project that could generate this problem, I started to have it exactly after updating my vcpkg.

**Environment**
- OS: Windows 11
- Compiler: Visual Studio 17.3

**To Reproduce**
Steps to reproduce the behavior:
Install dependencies to compile project Canary and before generating the executable the error will happen.
https://github.com/opentibiabr/canary

**Expected behavior**
Compile normally.

**Failure logs**
```
[148/149] Building CXX object src\CMakeFiles\canary.dir\server\network\protocol\protocolgame.cpp.obj
   [149/149] Linking CXX executable bin\canary.exe
   FAILED: bin/canary.exe
   cmd.exe /C "cd . && "C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin\cmake.exe" -E vs_link_exe --intdir=src \CMakeFiles\canary.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100220~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10 \bin\100220~1.0\x64\mt.exe --manifests -- C:\PROGRA~1\MICROS~1\2022\COMMUN~1\VC\Tools\MSVC\1434~1.319\bin\Hostx64\x64\ link.exe /nologo @CMakeFiles\canary.rsp /out:bin\canary.exe /implib:src\canary.lib /pdb:bin\canary.pdb /version:0.0 /machine:x64 /debug /INCREMENTAL /subsystem: console /INCREMENTAL:NO /LTCG && cmd.exe /C "cd /D G:\GitHub\otbr\canary-fork\build\windows-release\src && C:\Windows\System32\WindowsPowerShell\v1.0\powershell. exe -noprofile -executionpolicy Bypass -file G:/GitHub/vcpkg/scripts/buildsystems/msbuild/applocal.ps1 -targetBinary G:/GitHub/otbr/canary-fork/build/windows-release/bin/canary.exe -installedDir G:/GitHub/otbr/canary-fork/build/windows-release/vcpkg_installed/x64-windows-static/bi n -OutVariable out""
   LINK Pass 1: command "C:\PROGRA~1\MICROS~1\2022\COMMUN~1\VC\Tools\MSVC\1434~1.319\bin\Hostx64\x64\link.exe /nologo @CMakeFiles\canary.rsp /out:bin\canary.exe /implib:src\canary.lib /pdb:bin\canary.pdb /version:0.0 /machine:x64 /debug /INCREMENTAL /subsystem:console /INCREMENTAL:NO /LTCG /MANIFEST /MANIFESTFILE :src\CMakeFiles\canary.dir/intermediate.manifest src\CMakeFiles\canary.dir/manifest.res" failed (exit code 1169) with the following output:
G:\GitHub\otbr\canary-fork\build\windows-release\fileloader.cpp.obj : error LNK2005: "class std::basic_string<wchar_t,struct std::char_traits<wchar_t>,class std::allocator< wchar_t> > __cdecl mio::detail::win::s_2_ws(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (?s_2_ws@win @detail@mio@@YA?AV?$basic_string@_WU?$char_traits@_W@std@@V?$allocator@_W@2@@std@@AEBV?$basic_string@DU?$char_traits@D@std@ @V?$allocator@D@2@@5@@Z) already defined in cmake_pch.cxx.obj
  
      Creating src\canary.lib library and src\canary.exp object
  
G:\GitHub\otbr\canary-fork\build\windows-release\bin\canary.exe : fatal error LNK1169: Found one or more defined multiple symbols
  
   ninja: build stopped: subcommand failed.
```

```
G:\GitHub\otbr\canary-fork\build\windows-release\fileloader.cpp.obj : error LNK2005: "class std::basic_string<wchar_t,struct std::char_traits<wchar_t>,class std::allocator< wchar_t> > __cdecl mio::detail::win::s_2_ws(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (?s_2_ws@win @detail@mio@@YA?AV?$basic_string@_WU?$char_traits@_W@std@@V?$allocator@_W@2@@std@@AEBV?$basic_string@DU?$char_traits@D@std@ @V?$allocator@D@2@@5@@Z) already defined in cmake_pch.cxx.obj
```
The same symbol was found in `__cdecl mio::detail::win::s_2_ws`, please check the link list.

--- a/ports/mio/portfile.cmake
+++ b/ports/mio/portfile.cmake
@@ -2,8 +2,8 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO mandreyel/mio
-    REF 3f86a95c0784d73ce6815237ec33ed25f233b643
-    SHA512 18bbc41d5c3b29ecafe19cef29687380d8f4f27279af08edb0d5e65ee1d71162f3cf75d8efaa324cd11301f3f49fadaec3e4c1514607d5065ddd7a65bf32ee2a
+    REF 8b6b7d878c89e81614d05edca7936de41ccdd2da
+    SHA512 444131d4839f2244dd88722f5bfad2cfa47336e2a4405518a2ff8f0d80f2755321d7d627f8d5b890864a5dc3f3f810a1c7dd6588ff3e9039a6ef7d010e0f2f06
     HEAD_REF master
 )
 
@@ -20,4 +20,4 @@ vcpkg_cmake_config_fixup(CONFIG_PATH share/cmake/mio)
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug")
 
 # Handle copyright
-file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
+vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE")

--- a/ports/mio/vcpkg.json
+++ b/ports/mio/vcpkg.json
@@ -1,6 +1,6 @@
 {
   "name": "mio",
-  "version-date": "2023-01-25",
+  "version-date": "2023-03-03",
   "description": "Cross-platform header-only C++11 library for memory mapped file IO.",
   "homepage": "https://github.com/mandreyel/mio",
   "license": "MIT",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -5165,7 +5165,7 @@
       "port-version": 0
     },
     "mio": {
-      "baseline": "2023-01-25",
+      "baseline": "2023-03-03",
       "port-version": 0
     },
     "mlpack": {

--- a/versions/m-/mio.json
+++ b/versions/m-/mio.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "32e1d9bb77df214fec0e6ea240671114e3d3df30",
+      "version-date": "2023-03-03",
+      "port-version": 0
+    },
+    {
       "git-tree": "a71deb82d263c979ed9077747778cc2295baee36",
       "version-date": "2023-01-25",
       "port-version": 0

